lib/galaxy/webapps/galaxy/controllers/workflow.py

```diff
@@ -657,6 +657,9 @@ def editor(self, trans, id=None, workflow_id=None, version=None):
             if workflow_id:
                 workflow = trans.sa_session.query(model.Workflow).get(trans.security.decode_id(workflow_id))
                 stored_workflow = workflow.stored_workflow
+                if stored_workflow is None:
```

Comment:
I think we might be better off calling `self.app.workflow_manager.get_stored_workflow` instead of line 658-661 to reduce the duplication. And we should also try to do this when importing a shared subworkflow (but that's a bonus that's not necessary for the fix)

Reply:
You are absolutely right about the duplication.
I thought doing this somewhat lazy StoredWorkflow creation when actually trying to edit was a feature :laughing: do we want to create it right away after importing then?

Reply:
Yep, that sounds like the right thing to do

Reply:
Disagreeing with @mvdbeek is always against my better judgement but I think I do in fact disagree. You import a workflow with subworkflows and now you have mysterious new workflows in the workflow list and you don't know why - this seems counter intuitive and annoying. So I kind of disagree that we should auto-create those. I think of the storedworkflow as the entry in the list and I don't think you want those generally for these subworkflows. After editing it - sure - this PR seems to be a good step if that functionality is there.

Reply:
They are hidden

Reply:
:popcorn: :eyes: